tls/extensions/s2n_client_supported_groups.c

```diff
@@ -160,7 +163,7 @@ static int s2n_choose_supported_group(struct s2n_connection *conn)
      * populated with anything. */
     for (size_t i = 0; i < kem_pref->tls13_kem_group_count; i++) {
         const struct s2n_kem_group *candidate_kem_group = conn->kex_params.mutually_supported_kem_groups[i];
-        if (candidate_kem_group != NULL) {
+        if (candidate_kem_group != NULL && s2n_kem_group_is_available(candidate_kem_group)) {
```

Comment:
I guess this wouldn't work here though because you're iterating over the mutually supported array and not the preference list. So maybe it wouldn't be as useful as I was thinking..

Reply:
hm, `s2n_kem_preferences_get_available_group` could return `NULL` in the case of an unavailable group, right? if so, you'd still have to handle that with a null check, introducing a conditional branch. i'm not sure how much that saves us vs. just checking availability in the loop.

Reply:
It would make it harder to forget to check availability. I'd say if you can use a common method in enough places, it's probably worth it. But I'm not sure I can really picture the change, so you might need to just see how it looks.